### ISSUE TITLE
Terraform not prompting for missing variables when `-var` is used

### ISSUE THREAD

**Issue by ajcrites** [hashicorp/terraform#7975]
If you use `-var` with terraform commands (at least `plan`, `apply`, and `destroy`) and you do not specify all variables, terraform will not prompt for the missing variables.
### Terraform Version

0.7
### Affected Resource(s)

core
### Terraform Configuration Files

```
variable "FOO" {}
variable "BAR" {}

resource "null_resource" "dummy-echo" {
    provisioner "local-exec" {
        command = "echo ${var.FOO} ${var.BAR}"
    }
}
```
### Debug Output

Run with `terraform plan -var 'FOO=baz'`

```
There are warnings and/or errors related to your configuration. Please
fix these before continuing.

Errors:

  * 1 error(s) occurred:

* Required variable not set: BAR
```
### Expected Behavior

Terraform should prompt for missing variables in all contexts
### Actual Behavior

If you use `-var` but do not specify _all_ variables (either with defaults, another `-var-file`, or `terraform.tfvars`) Terraform will not prompt for any missing variables and report an error instead.
### Steps to Reproduce

`terraform plan -var 'FOO=bar'` with the sample config is enough.
### References

Related to #6554 (no prompt for other `.tfvars` file), #1230 and #1899

**Comment by mitchellh** [comment]
Reproduced. This should really only happen if `-input=false` but input is true (both by default and forced) and I still get this error. Will look at it.

**Comment by mitchellh** [comment]
Fix queued up in PR #9794!
